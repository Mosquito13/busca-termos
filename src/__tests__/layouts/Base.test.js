import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';

import BaseLayout from '../../layouts/Base';

const getStateMock = (darkTheme, compactLayout) => ({
  Settings: {
    darkTheme,
    compactLayout
  },
  Frame: {
    maximized: false
  }
});

const prepareMocks = (dispatchMock, stateMock) => {
  jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock);
  jest
    .spyOn(reactRedux, 'useSelector')
    .mockImplementation((selector) => selector(stateMock));
};

describe(`${BaseLayout.name}`, () => {
  describe('snapshots', () => {
    it('should render normal layout and light theme', () => {
      prepareMocks(jest.fn(), getStateMock(false, false));

      const tree = renderer.create(<BaseLayout />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render normal layout and dark theme', () => {
      prepareMocks(jest.fn(), getStateMock(true, false));

      const tree = renderer.create(<BaseLayout />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render compact layout and light theme', () => {
      prepareMocks(jest.fn(), getStateMock(false, true));

      const tree = renderer.create(<BaseLayout />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render compact layout and dark theme', () => {
      prepareMocks(jest.fn(), getStateMock(true, true));

      const tree = renderer.create(<BaseLayout />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
