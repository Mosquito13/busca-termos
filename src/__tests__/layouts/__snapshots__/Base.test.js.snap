// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout snapshots should render compact layout and dark theme 1`] = `
<div
  className="base-layout base-layout--compact theme--dark"
>
  <div
    className="base-layout__frame"
  >
    <div
      className="app-frame"
    >
      <div
        className="app-frame__title"
      >
        <div
          className="app-frame__title-icon"
          onDoubleClick={[Function]}
        >
          <svg>
            icon.svg
          </svg>
        </div>
      </div>
      <div
        className="app-frame__buttons"
      >
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 8v1H3V8h11z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v10h10V3H3zm9 9H4V4h8v8z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button app-frame__button--close"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="base-layout__content"
  />
</div>
`;

exports[`BaseLayout snapshots should render compact layout and light theme 1`] = `
<div
  className="base-layout base-layout--compact theme--light"
>
  <div
    className="base-layout__frame"
  >
    <div
      className="app-frame"
    >
      <div
        className="app-frame__title"
      >
        <div
          className="app-frame__title-icon"
          onDoubleClick={[Function]}
        >
          <svg>
            icon.svg
          </svg>
        </div>
      </div>
      <div
        className="app-frame__buttons"
      >
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 8v1H3V8h11z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v10h10V3H3zm9 9H4V4h8v8z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button app-frame__button--close"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="base-layout__content"
  />
</div>
`;

exports[`BaseLayout snapshots should render normal layout and dark theme 1`] = `
<div
  className="base-layout theme--dark"
>
  <div
    className="base-layout__frame"
  >
    <div
      className="app-frame"
    >
      <div
        className="app-frame__title"
      >
        <div
          className="app-frame__title-icon"
          onDoubleClick={[Function]}
        >
          <svg>
            icon.svg
          </svg>
        </div>
      </div>
      <div
        className="app-frame__buttons"
      >
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 8v1H3V8h11z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v10h10V3H3zm9 9H4V4h8v8z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button app-frame__button--close"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="base-layout__content"
  />
</div>
`;

exports[`BaseLayout snapshots should render normal layout and light theme 1`] = `
<div
  className="base-layout theme--light"
>
  <div
    className="base-layout__frame"
  >
    <div
      className="app-frame"
    >
      <div
        className="app-frame__title"
      >
        <div
          className="app-frame__title-icon"
          onDoubleClick={[Function]}
        >
          <svg>
            icon.svg
          </svg>
        </div>
      </div>
      <div
        className="app-frame__buttons"
      >
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 8v1H3V8h11z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 3v10h10V3H3zm9 9H4V4h8v8z"
              />
            </svg>
          </div>
        </button>
        <button
          className="app-frame__button app-frame__button--close"
          onClick={[Function]}
        >
          <div
            className="app-frame__button-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="base-layout__content"
  />
</div>
`;
